### PR TITLE
Juniper methods support defaulted interfaces

### PR DESCRIPTION
**Pull request by fbouliane**
The behavior of juniper switches when the configuration is default
is to return no configuration node when querying. The new fake switches
1.0.29 introduce this change of behavior. This prepares the terrain for
this new version.

Add test so netman supports default configuration on all methods.
Add test so netman still raises on Unknown interface on a per-method
basis.

### CODE REVIEW

File: netman/adapters/switches/juniper/base.py [hunk 1]
```diff
@@ -215,15 +215,20 @@ def set_access_mode(self, interface_id):
         if len(update_attributes) > 0:
             update = Update()
             update.add_interface(interface_update(interface_id, "0", update_attributes))
-
-            self._push(update)
+            try:
```
Comment (lindycoder) [inline netman/adapters/switches/juniper/base.py]: can you put this try except in a _push_interface_update that would always handle the interface not found error?

Comment (fbouliane) [inline netman/adapters/switches/juniper/base.py]: Comment adressed.

File: tests/adapters/switches/juniper_test.py [hunk 2]
```diff
@@ -2553,7 +2732,7 @@ def test_unset_interface_access_vlan_on_trunk_mode_raises(self):
 
         assert_that(str(expect.exception), contains_string("Operation cannot be performed on a trunk mode interface"))
 
-    def test_unset_interface_access_vlan_on_unknown_interface_raises(self):
+    def test_unset_interface_access_vlan_on_default_interface_works(self):
```
Comment (lindycoder) [inline tests/adapters/switches/juniper_test.py]: Didn't you remove a valid test here, shouldn't the default interface be a separate test?

File: tests/adapters/switches/juniper_test.py [hunk 3]
```diff
@@ -3002,7 +3238,7 @@ def test_unset_interface_native_vlan_when_none_is_set_raises(self):
 
         assert_that(str(expect.exception), contains_string("Trunk native Vlan is not set on interface ge-0/0/6"))
 
-    def test_unset_interface_native_vlan_on_unknown_interface_raises(self):
+    def test_unset_interface_native_vlan_on_default_interface_raises(self):
```
Comment (lindycoder) [inline tests/adapters/switches/juniper_test.py]: same comment

**Comment by lindycoder** [comment]
👍

File: netman/adapters/switches/juniper/base.py [hunk 4]
```diff
@@ -215,15 +215,16 @@ def set_access_mode(self, interface_id):
...
     def set_trunk_mode(self, interface_id):
         update_attributes = []
 
-        interface = self.get_interface(interface_id)
+        config = self.query(one_interface(interface_id), all_vlans)
+        interface_node = self.get_interface_config(interface_id, config)
+        interface = self.node_to_interface(interface_node, config)
 
-        if interface.port_mode is ACCESS:
+        if interface.port_mode is ACCESS or interface.port_mode is None:
```
Comment (mat128) [inline netman/adapters/switches/juniper/base.py]: I don't see how this relates to the change?

File: netman/adapters/switches/juniper/base.py [hunk 5]
```diff
@@ -1101,20 +1124,24 @@ def _compute_edge_state_modifications(interface_id, edge, config):
...
-            if edge_node is None:
-                modifications.append("<edge />")
-            if no_root_port_node is None:
-                modifications.append("<no-root-port />")
-        elif edge is False:
-            if edge_node is not None:
-                modifications.append("<edge operation=\"delete\" />")
-            if no_root_port_node is not None:
-                modifications.append("<no-root-port operation=\"delete\" />")
+    if edge is True:
```
Comment (mat128) [inline netman/adapters/switches/juniper/base.py]: This can happen without having an rstp_node?

**Comment by mat128** [comment]
Questions inline, -1 for now.

File: netman/adapters/switches/juniper/base.py (hunk 4 above)
Comment (mat128) [inline netman/adapters/switches/juniper/base.py]: got it, none is default, access is access.

File: netman/adapters/switches/juniper/base.py (hunk 5 above)
Comment (mat128) [inline netman/adapters/switches/juniper/base.py]: OK, new code creates the rstp_node if it doesnt exist, whereas before setting edge would just noop. Nice catch!

**Pull request merged**